internal/report.go

```diff
@@ -17,5 +31,66 @@ type Report struct {
 	// Network error.
 	Error error `json:"error,omitempty"`
 	// Extra information. Depends on the protocol.
-	Extra string `json:"extra,omitempty"`
+	Extra  string `json:"extra,omitempty"`
+	Format Format
+}
+
+func (r *Report) PrintFormatted(f Format) {
+
+	switch f {
+	case HumanFormat:
+		printHumanFormat(r)
+	case JSONFormat:
+		printJSONFormat(r)
+	case GrepFormat:
+		printGrepableFormat(r)
+	}
+
+}
+func printJSONFormat(r *Report) {
```

Comment:
I would made these part of the `Report` definition: `func (r *Report) printJSONFormat() {`

Reply:
Like before, this one should also return a `string`.

Reply:
So I would also rename these to: `NewLine()`, `newLineJSON`, `newLineGrep`, etc.